packages/xo-server/src/xapi/index.js

```diff
@@ -1437,7 +1437,25 @@ export default class Xapi extends XapiBase {
     return /* await */ this._snapshotVm(this.getObject(vmId), nameLabel)
   }
 
-  async _startVm(vm, host, force) {
+  async _startVm(vm, host, force = false) {
+    if (!force) {
+      const runningVmsVifsByMac = groupBy(
+        flatMap(
```

Comment:
You can increase [Node version requirement](https://github.com/vatesfr/xen-orchestra/blob/aff874c68a9256f44db8ac611ba14191db39325a/packages/xo-server/package.json#L32) and use [native `Array#flatMap`](https://node.green/#ES2019-features-Array-prototype--flat--flatMap--Array-prototype-flatMap).